tools/ensure-upx.ps1

```
@@ -0,0 +1,219 @@
+function Find-Upx {
+    $cmd = Get-Command upx -ErrorAction SilentlyContinue
+    if ($cmd) { return $cmd.Source }
+    return $null
+}
+
+# ---------------------------------------------------------------------------
+# Helper: refresh the process PATH from the machine + user environment so
+# a freshly installed binary (winget/choco) becomes visible without starting
+# a new shell.
+# ---------------------------------------------------------------------------
+function Update-ProcessPath {
```

PSScriptAnalyzer (1.25.0)

[warning] Function 'Update-ProcessPath' has verb that could change system state. Therefore, the function has to support 'ShouldProcess'.

(PSUseShouldProcessForStateChangingFunctions)